src/julia-syntax.scm

```diff
@@ -3451,23 +3477,40 @@ f(x) = yt(x)
 ;; collect all toplevel-butfirst expressions inside `e`, and return
 ;; (ex . stmts), where `ex` is the expression to evaluated and
 ;; `stmts` is a list of statements to move to the top level.
+;; also expand set-binding-type expressions and handle them similarly
```

Comment:
Looks like this still allows global type declarations in local scopes? Jameson and I agree we can just disallow that, which should simplify the code as well.

Reply:
Ok, I have now disallowed that. Something like `f() = (global x = 1;)` still freezes the type of `x` to `Any` (unless already set) before `f` is ever executed though in order not to introduce dynamic calls to `convert`.